proposal/0000-record-dot-syntax.md

```diff
@@ -12,77 +12,276 @@ the link, and delete this bold sentence.**
+
+data Taken =
+  Taken { year : Int
+        , term : Quarter
+        }
+
+data Class =
+  Class { hours : Int
+        , units : Int
+        , grade : Grade
+        , result : Status
+        , taken : Taken
+        }
+
+getResult :: Class -> Status
+getResult c = c.result -- get
+
+setResult :: Class -> Status -> Class
+setResult c r = c{result = r} -- update
+
+setYearTaken :: Class -> Int -> Class
+setYearTaken c y = c{taken.year = y} -- nested update
+
+addYears :: Class -> Int -> Class
+addYears c n = c{taken.year + n} -- update via op
+
+squareUnits :: Class -> Class
+squareUnits c = c{units & (\x -> x * x)} -- update via function
+
+getResults :: [Class] -> [Status]
+getResults = map (.result) -- section
+
+getTerms :: [Class]  -> [Quarter]
+getTerms = map (.taken.term) -- nested section
+```
+
+A full, rigorous set of examples (as tests) are available in the examples directory of [this repository](https://github.com/ndmitchell/record-dot-preprocessor). Those tests include infix applications, polymorphic data types, interoperation with other extensions and more. They follow the [specifications given earlier](#proposed-change-specification).
 
 ## Effect and Interactions
 
-Detail how the proposed change addresses the original problem raised in the
-motivation.
+**Polymorphic updates:** When enabled, this extension takes the `a{b=c}` syntax and uses it to mean `setField`. The biggest difference a user is likely to experience is that the resulting type of `a{b=c}` is the same as the type `a` - you _cannot_ change the type of the record by updating its fields. The removal of polymorphism is considered essential to preserve decent type inference, and is the only option supported by [the `HasField` proposal](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0158-record-set-field.rst).
 
-Discuss possibly contentious interactions with existing language or compiler
-features.
+**Stealing a.b syntax:** The `a.b` syntax is commonly used in conjunction with the `lens` library, e.g. `expr^.field1.field2`. Treating `a.b` without spaces as a record projection would break such code. The alternatives would be to use a library with a different lens composition operator (e.g. `optics`), introduce an alias in `lens` for `.` (perhaps `%`), write such expressions with spaces, or not enable this extension when also using lenses. While unfortunate, we consider that people who are heavy users of lens don't feel the problems of inadequate records as strongly, so the problems are lessened.
 
+**Enabled extensions:** When `RecordDotSyntax` is enabled it should imply the `NoFieldSelectors` extension and allow duplicate record field labels. It would be possible for `RecordDotSyntax` to imply `DuplicateRecordFields`, but we suspect that if people become comfortable with `RecordDotSyntax` then there will be a desire to remove the `DuplicateRecordFields` extension, so we don't want to build on top of it.
```

Comment:
I don’t understand why this has to imply `NoFieldSelectors`. Given that this is builtin syntax why does it matter if there are top-level field selectors or not?

Reply:
It doesn't have to. I'll add that as a note in the proposal.